sematic/db/models/resolution.py

```diff
@@ -0,0 +1,177 @@
+            from_status = ResolutionStatus[from_status]
+        if not isinstance(to_status, ResolutionStatus):
+            to_status = ResolutionStatus[to_status]
+        return to_status in _ALLOWED_TRANSITIONS[from_status]
+
+
+_ALLOWED_TRANSITIONS = {
+    ResolutionStatus.CREATED: {ResolutionStatus.SCHEDULED, ResolutionStatus.FAILED},
+    ResolutionStatus.SCHEDULED: {ResolutionStatus.RUNNING, ResolutionStatus.FAILED},
+    ResolutionStatus.RUNNING: {ResolutionStatus.COMPLETE, ResolutionStatus.FAILED},
+    ResolutionStatus.COMPLETE: {},
+    ResolutionStatus.FAILED: {},
+}
+
+
+@unique
+class ResolutionKind(Enum):
+    """The kind of compute used to execute the resolution"""
+
+    LOCAL = "LOCAL"  # for non-detached mode
+    KUBERNETES = "KUBERNETES"  # for detached mode
+
+
+class Resolution(Base, JSONEncodableMixin):
+    """Represents a session of a resolver
+
+    Attributes
+    ----------
+    root_id:
+        The id of the root run which this resolution is resolving.
+    status:
+        The state of the resolver session, see ResolutionStatus.
+    kind:
+        The kind of resolver session (ex: on k8s or not).
+    docker_image_uri:
+        The docker image URI for the resolution. May be null when
+        doing a non-detached (local) resolution
+    settings_env_vars:
+        The Sematic settings from the user's environment for the user
+        who launched this resolution.
+    """
+
+    __tablename__ = "resolutions"
+
+    root_id: str = Column(
+        types.String(),
+        nullable=False,
+        primary_key=True,
+        default=lambda: uuid.uuid4().hex,
+    )
+    status: ResolutionStatus = Column(  # type: ignore
+        types.String(), nullable=False, info={ENUM_KEY: ResolutionStatus}
+    )
+    kind: ResolutionKind = Column(  # type: ignore
+        types.String(), nullable=False, info={ENUM_KEY: ResolutionStatus}
+    )
+    docker_image_uri: str = Column(types.String(), nullable=True, default=None)
+    settings_env_vars: Dict[str, str] = Column(
+        types.JSON, nullable=False, default=lambda: {}
+    )
+
+    @validates("status")
+    def validate_status(self, key, value) -> str:
+        """
+        Validates that the status value is allowed.
+        """
+        if isinstance(value, ResolutionStatus):
+            return value.value
+
+        try:
+            return ResolutionStatus[value].value
+        except Exception:
+            raise ValueError("status must be a ResolutionStatus, got {}".format(value))
+
+    @validates("kind")
+    def validate_kind(self, key, value) -> str:
+        """
+        Validates that the kind value is allowed.
+        """
+        if isinstance(value, ResolutionKind):
+            return value.value
+
+        try:
+            return ResolutionKind[value].value
+        except Exception:
+            raise ValueError("kind must be a ResolutionKind, got {}".format(value))
+
+    def check_update(self, other: "Resolution") -> Optional[str]:
```

Comment:
Can we do something like
```python
existing_resolution.update_with(resolution)
save_resolution(existing_resolution)
```
that updates `existing_resolution`'s fields. And that would raise an exception for fields that are not valid.

Reply:
yup, that makes sense